src/mscorlib/src/System/String.Comparison.cs

```diff
@@ -1144,5 +1144,16 @@ public override int GetHashCode()
 
             return referenceCulture.CompareInfo.IsPrefix(this, value, ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None);
         }
+
+        [Pure]
+        public Boolean StartsWith(Char value)
```

Comment:
I can't see any difference from a normal property. It gets inlined (as any simple property)

I created the following test to validate it, but couldn't get any conclusion from that: https://gist.github.com/fujiy/284a31a3237720c8f9e895844a426698

I also did the same tests running on .NET 4.6.2. Interesting, the field returning method get a lot more code in .NET Core